UI/DropsController.gd

```diff
@@ -15,16 +15,30 @@ func _ready():
 	_update_drops_tables()
 	
 func _update_drops_tables() -> void:
-	_clear_drops_tables()
+	var has_filter_text = $DropsFilterLineEdit != null and !$DropsFilterLineEdit.text.empty()
+	_clear_drops_tables(has_filter_text)
+	
 	for drops_table in SetProvider.get_all_drops_tables():
 		if drops_table.id < 0:
 			push_error("Found drops table with ID -1. Ignoring.")
 		else:
 			$HFlowContainer/DropsTableOptionButton.add_item(drops_table.name, drops_table.id)
+	
+	# select the currently selected drops table if it exists
+	if _selected_drops_table != null:
+		var selected_option_idx: int = $HFlowContainer/DropsTableOptionButton.get_item_index(_selected_drops_table.id)
+		if selected_option_idx >= 0:
+			$HFlowContainer/DropsTableOptionButton.select(selected_option_idx)
+
 		
-func _clear_drops_tables() -> void:
+func _clear_drops_tables(has_filter_text: bool) -> void:
 	$HFlowContainer/DropsTableOptionButton.clear()
-	$HFlowContainer/DropsTableOptionButton.add_item("-- No Drops --", NO_DROPS_OPTION_ID)
+	# Add the "-- No Drops --" option only if there's no filter text
+	if not has_filter_text:
+		$HFlowContainer/DropsTableOptionButton.add_item("-- No Drops --", NO_DROPS_OPTION_ID)
+	if _selected_drops_table != null:
```

Comment:
By this point there'd be no elements in the drop-down since they all have been cleared. This wouldn't achieve anything